src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,46 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> head;
+    private int size = 0;
 
+    public void push(T value) {
+        Node<T> node = new Node<T>(value, head);
+        head = node;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (head == null) {
+            throw new EmptyStackException();
+        }
+        T value = head.value;
+        return value;
```

Comment:
Do not create redundant variables, just return it when it's possible
```suggestion
        return head.value;
```